eventlogger/hapi_server.js

```diff
@@ -47,14 +52,19 @@ module.exports.watch = function  (logger, server, options) {
     const level = tags.debug ? 'debug': 'info';
     logger[level]({ req: request, data: data.data });
   }).on('request-error', function (request, error) {
+    let payload = request.pre && request.pre._originalPayload || request.payload;
+    if (obfuscatePayload) {
+      payload = obfuscator.obfuscateSafe(payload);
+    }
+    if (stringifyPayload) {
+      payload = JSON.stringify(payload);
```

Comment:
Is it safe to assume `JSON.stringify` always gets well formed data?

Reply:
I think it's pretty safe, but we could make this even safer. `JSON.stringify()` would throw when the argument contains circular/cyclical references. If we look [under the hood](https://github.com/trentm/node-bunyan/blob/fe31b83e42d9c7f784e83fdcc528a7c76e0dacae/lib/bunyan.js#L1200-L1230), bunyan is also stringifying the log entry, but does so in a safer way, by falling back to the `safe-json-stringify` module (which basically detects cycles).

The request payload should not have any circular references in normal circumstances. But it could happen if the application mutates it (which is bad imo).

For now, I suggest we leave like this, but please let me know if you feel otherwise.